http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/util/AKSInterpreter.scala

```diff
@@ -782,31 +787,56 @@ class AKSInterpreter[F[_]](config: AKSInterpreterConfig,
       for {
         ctx <- ev.ask
         wsmApi <- buildWsmControlledResourceApiClient
-        controlledDbsForApp = appInstall.databases.collect { case d @ ControlledDatabase(_, _, _) => d }
-        // retrieve databases that might already be created in workspace
+
+        // get a list of databases needed for this app
+        controlledDbsRequiredForApp = appInstall.databases.collect { case d @ ControlledDatabase(_, _, _) => d }
+
+        // retrieve set of info about any WSM existing databases for this workspace
         existingControlledDbsInWorkspace <- retrieveWsmDatabases(wsmResourceApi,
-                                                                 controlledDbsForApp.map(_.prefix).toSet,
+                                                                 controlledDbsRequiredForApp.map(_.prefix).toSet,
                                                                  workspaceId.value
         )
-        wsmControlledDBResources <- controlledDbsForApp
+        wsmControlledDBResources <- controlledDbsRequiredForApp
           .map { controlledDbForApp =>
+            val maybeExistingDb = existingControlledDbsInWorkspace
+              .find(existingDb => controlledDbForApp.prefix == existingDb.wsmDatabaseName)
```

Comment:
I assume (hope 😬 ) that the DB names are enforced to be unique within a workspace

Reply:
They are -- WSM names are unique within a workspace.